feat(templates): add SMTP configuration to ente-photos compose templates

ariqpradipa opened the pull request:
Add environment variables for SMTP host, port, username, password, email, and sender name to enable email functionality in the museum service. This enhances the ente-photos setup with email capabilities for notifications.

## Changes
- add smtp env configuration in ente-photos-with-s3 and ente-photos

## Issues
- fix #6882

ariqpradipa committed:
feat(templates): add SMTP configuration to ente-photos compose templates

Add environment variables for SMTP host, port, username, password, email, and sender name to enable email functionality in the museum service. This enhances the ente-photos setup with email capabilities for notifications or other features.

ShadowArcanist commented on templates/compose/ente-photos-with-s3.yaml:
```diff
@@ -7,7 +7,7 @@
 services:
   museum:
-    image: 'ghcr.io/ente-io/server:613c6a96390d7a624cf30b946955705d632423cc' # Released at 2025-09-14T22:16:37-07:00
+    image: 'ghcr.io/ente-io/server:613c6a96390d7a624cf30b946955705d632423cc'
```
What is the reason for the removal of the release date and time from the comments here?

ariqpradipa replied:
No particular reason. I copied the compose file into Coolify Docker Compose and used Coolify formatting. Somehow, it removed the comments. I’ll add them back.

ShadowArcanist commented on templates/compose/ente-photos-with-s3.yaml:
```diff
@@ -46,9 +52,8 @@ services:
-      
   web:
-    image: 'ghcr.io/ente-io/web:ca03165f5e7f2a50105e6e40019c17ae6cdd934f' # Released at 2025-10-08T00:57:05-07:00
+    image: 'ghcr.io/ente-io/web:ca03165f5e7f2a50105e6e40019c17ae6cdd934f'
```
What is the reason for the removal of the release date and time from the comments here?

ariqpradipa replied:
No particular reason. I copied the compose file into Coolify Docker Compose and used Coolify formatting. Somehow, it removed the comments. I’ll add them back.

ShadowArcanist commented on templates/compose/ente-photos-with-s3.yaml:
```diff
@@ -80,9 +84,8 @@ services:
-      
   minio:
-    image: 'quay.io/minio/minio:RELEASE.2025-09-07T16-13-09Z' # Released at 2025-09-07T16-13-09Z
+    image: 'quay.io/minio/minio:RELEASE.2025-09-07T16-13-09Z'
```
What is the reason for the removal of the release date and time from the comments here?

ariqpradipa replied:
No particular reason. I copied the compose file into Coolify Docker Compose and used Coolify formatting. Somehow, it removed the comments. I’ll add them back.

ShadowArcanist commented on templates/compose/ente-photos-with-s3.yaml:
```diff
@@ -100,10 +103,9 @@ services:
       
   minio-init:
-    image: 'minio/mc:RELEASE.2025-08-13T08-35-41Z' # Released at 2025-08-13T08-35-41Z
+    image: 'minio/mc:RELEASE.2025-08-13T08-35-41Z'
```
What is the reason for the removal of the release date and time from the comments here?

ariqpradipa replied:
No particular reason. I copied the compose file into Coolify Docker Compose and used Coolify formatting. Somehow, it removed the comments. I’ll add them back.

ShadowArcanist commented on templates/compose/ente-photos-with-s3.yaml:
```diff
@@ -28,6 +28,12 @@ services:
+      - 'ENTE_SMTP_USERNAME=${ENTE_SMTP_USERNAME}'
+      - 'ENTE_SMTP_PASSWORD=${ENTE_SMTP_PASSWORD}'
+      - 'ENTE_SMTP_EMAIL=${ENTE_SMTP_EMAIL}'
+      - 'ENTE_SMTP_SENDER_NAME=${ENTE_SMTP_SENDER_NAME}'
```
Add Encryption method env
```suggestion
      - 'ENTE_SMTP_SENDER_NAME=${ENTE_SMTP_SENDER_NAME}'
      - 'ENTE_SMTP_ENCRYPTION=${ENTE_SMTP_ENCRYPTION:-tls}'
```

ShadowArcanist commented on templates/compose/ente-photos.yaml:
```diff
@@ -39,6 +39,13 @@ services:
+      - ENTE_SMTP_PORT=${ENTE_SMTP_PORT:-587}
+      - ENTE_SMTP_USERNAME=${ENTE_SMTP_USERNAME}
+      - ENTE_SMTP_PASSWORD=${ENTE_SMTP_PASSWORD}
+      - ENTE_SMTP_EMAIL=${ENTE_SMTP_EMAIL}
+      - ENTE_SMTP_SENDER_NAME=${ENTE_SMTP_SENDER_NAME}
+
```
Add Encryption method env
```suggestion
      - 'ENTE_SMTP_SENDER_NAME=${ENTE_SMTP_SENDER_NAME}'
      - 'ENTE_SMTP_ENCRYPTION=${ENTE_SMTP_ENCRYPTION:-tls}'
```

ariqpradipa committed:
feat(templates): add SMTP encryption configuration to ente-photos compose templates

Add optional ENTE_SMTP_ENCRYPTION environment variable with default value 'tls' to museum service in both ente-photos and ente-photos-with-s3 templates. This allows users to configure SMTP encryption settings for email functionality.

andrasbacsai commented:
Thank you for the PR!
